delivery_purchase_label/models/mail_compose_message.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, models
+
+
+class MailComposer(models.TransientModel):
+    _inherit = "mail.compose.message"
+
+    @api.model
+    def generate_email_for_composer(self, template_id, res_ids, fields):
+        res = super().generate_email_for_composer(template_id, res_ids, fields)
+        if self.model != "purchase.order":
+            return res
+        if (
+            template_id
+            in self.env["purchase.order"]._mail_templates_to_not_attach_labels()
+        ):
+            return
+        if isinstance(res_ids, int):
+            res_ids = [res_ids]
```

Comment:
```suggestion
```

This doesn't seem necessary

given `self.env["purchase.order"].browse(1) == self.env["purchase.order"].browse([1])`